pkg/catalogv2/oci/oci.go

```diff
@@ -186,11 +189,17 @@ func GenerateIndex(ociClient *Client, URL string, credentialSecret *corev1.Secre
 
 				if maxTag != nil {
 					ociClient.tag = maxTag.Original()
-
-					// fetch the chart.yaml for the latest tag and add it to the index.
 					err = addToHelmRepoIndex(*ociClient, indexFile, orasRepository)
 					if err != nil {
-						return fmt.Errorf("failed to add tag %s in OCI repository %s to helm repo index: %w", maxTag.String(), ociClient.repository, err)
+						// Users can have access to only some repositories and not all.
+						// So, if pulling the chart fails due to Forbiden error, then skip it.
+						var errResp *errcode.ErrorResponse
+						if errors.As(err, &errResp) && errResp.StatusCode == http.StatusForbidden {
+							delete(indexFile.Entries, chartName)
+							logrus.Errorf("failed to add OCI repository %s to helm repo index: %v", ociClient.repository, err)
```

Comment:
IMO, this is not an error; it would be a warning since it is an expected situation. I don't know if Rancher has a log message policy.